Adds more EMAG effects to general machinery.

TerryTehCoder opened the pull request:
## About The Pull Request
Scrambling Security Camera C-Tags. The List of Cameras for A.I and other applications will contain scrambled names for the associated Security Cameras.

Emagged Lighters will always burn your fingers, and are hotter.

Cryocells can be sabotaged to function regardless of a beakers presence, silencing automatic alerts and stopping ejection.

You can now Slam someones head into a grill. Emagging a grill makes the flame hotter, and the damage better.

The Organ Smartfridge can be sabotaged to stop freezing organs, allowing organ decay even while organs are inside.

A.I Cores can now be emagged. Inciting a central command announcement, or a random scrambled lawset from any applicable laws.

Oh, and Runtime now says funny things when you Emag her.


This is a work in progress. I plan to add more EMAG expansions, however i'm open to code changes and suggestions. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

The EMAG is a very general purpose utility traitor item, and iconic at that. I think it deserves as many uses as we can reasonably justify giving it. I've been told being able to aggressively grill people was also a highly desired thing.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

TerryTehCoder committed:
Merge pull request #1 from BeeStation/master

Updating Fork for PR's

TerryTehCoder committed:
Merge pull request #2 from Donkstation/master

Updating to recent code.

TerryTehCoder committed:
Adds more EMAG effects (11/7/21)

Scrambling Security Camera C-Tags. The List of Cameras for A.I and other applications will contain scrambled names for the associated Security Cameras.

Emagged Lighters will always burn your fingers, and are hotter.

Cryocells can be sabotaged to function regardless of a beakers presence, and silencing automatic alerts and stopping ejection.

You can now Slam someones head into a grill. Emagging a grill makes the flame hotter, and the damage better.

The Organ Smartfridge can be sabotaged to stop freezing organs, allowing organ decay even while organs are inside.

A.I Cores can now be emagged. Inciting a central command announcement, or a random scrambled lawset from any applicable laws.

Oh, and Runtime now says funny things when you Emag her.

Zonespace27 commented on code/game/machinery/camera/camera.dm:
```diff
@@ -38,6 +38,7 @@
 	var/alarm_on = FALSE
 	var/busy = FALSE
 	var/emped = FALSE  //Number of consecutive EMP's on this camera
+	var/emagged = 0 //Has a chance to deactivate or EMP the Camera. 
```
```suggestion
	var/emagged = FALSE //Has a chance to deactivate or EMP the Camera. 
```

Zonespace27 commented on code/game/machinery/camera/camera.dm:
```diff
@@ -166,6 +169,15 @@
+	if(!emagged)
+		to_chat(user, "<span class='notice'>You wave the sequencer over the security camera, scrambling its network tag.")
+		do_sparks(2, 5, src)
+		emagged = 1
```
```suggestion
		emagged = TRUE
```

Zonespace27 commented on code/game/objects/items/cigs_lighters.dm:
```diff
@@ -553,6 +553,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	slot_flags = ITEM_SLOT_BELT
 	var/lit = 0
 	var/fancy = TRUE
+	var/emagged = 0 //It's really the small things we don't think about.
```
```suggestion
	var/emagged = FALSE //It's really the small things we don't think about.
```

Zonespace27 commented on code/game/objects/items/cigs_lighters.dm:
```diff
@@ -736,6 +741,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
+/obj/item/lighter/emag_act(mob/user)
+	if(!emagged)
+		emagged = 1
```
```suggestion
		emagged = TRUE
```

TerryTehCoder committed:
Updates Camera.Dm Emagged Desc.

Correctly labels emagged function in camera.dm

Zonespace27 commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -37,6 +37,7 @@
 	var/escape_in_progress = FALSE
 	var/message_cooldown
 	var/breakout_time = 300
+	var/emagged = 0
```
```suggestion
	var/emagged = FALSE
```

Zonespace27 commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -57,6 +58,11 @@
 
+/obj/machinery/atmospherics/components/unary/cryo_cell/emag_act(mob/user)
+	if(!emagged)
+		emagged = 1
```
```suggestion
		emagged = TRUE
```

Zonespace27 commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -57,6 +58,11 @@
+/obj/machinery/atmospherics/components/unary/cryo_cell/emag_act(mob/user)
+	if(!emagged)
+		emagged = 1
+		to_chat(user, "<span class='notice'>You short out the cryotubes automatic alert and ejection systems.</span>")
```
```suggestion
		to_chat(user, "<span class='notice'>You short out the cryotube's automatic alert and ejection systems.</span>")
```

Zonespace27 commented on code/modules/food_and_drinks/kitchen_machinery/grill.dm:
```diff
@@ -15,6 +15,7 @@
 	var/obj/item/reagent_containers/food/grilled_item
 	var/grill_time = 0
 	var/datum/looping_sound/grill/grill_loop
+	var/emagged = 0 //Now THAT'S H O T. PHEW, FLAMES FOR EVERYONE. 
```
```suggestion
	var/emagged = FALSE //Now THAT'S H O T. PHEW, FLAMES FOR EVERYONE. 
```

Zonespace27 commented on code/modules/food_and_drinks/kitchen_machinery/grill.dm:
```diff
@@ -65,6 +66,29 @@
+			var/mob/living/carbon/C = user.pulling
+			user.visible_message("<span class = 'danger'>[user] slams [C]'s face into [src]!</span>")
+			if(emagged)
+				C.apply_damage(35, BURN, BODY_ZONE_HEAD) //HOT GRILL HOT GRILL
```
C could probably scream or something

Zonespace27 commented on code/modules/food_and_drinks/kitchen_machinery/grill.dm:
```diff
@@ -144,5 +168,11 @@
 
+/obj/machinery/grill/emag_act(mob/user)
+	if(!emagged)
+		emagged = 1
```
```suggestion
		emagged = TRUE
```

Zonespace27 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -19,6 +19,7 @@
 	var/allow_ai_retrieve = FALSE
 	var/list/initial_contents
 	var/visible_contents = TRUE
+	var/emagged = 0 //I don't use this for any fridge except organ storage, but I assume you'll want it in parent for future.
```
```suggestion
	var/emagged = FALSE //I don't use this for any fridge except organ storage, but I assume you'll want it in parent for future.
```

Zonespace27 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -452,6 +456,13 @@
 
+/obj/machinery/smartfridge/organ/emag_act(mob/user, obj/item/O)
+	if(!emagged)
+		emagged = 1
```
```suggestion
		emagged = TRUE
```

Zonespace27 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -452,6 +456,13 @@
+/obj/machinery/smartfridge/organ/emag_act(mob/user, obj/item/O)
+	if(!emagged)
+		emagged = 1
+		to_chat(user, "<span class='notice'>You disable the smartfridges organ preservation.")
```
```suggestion
		to_chat(user, "<span class='notice'>You disable the smartfridge's organ preservation.")
```

Zonespace27 reviewed (commented):
just a quick look, will do more in depth review later

Zonespace27 commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -91,6 +91,22 @@
+	"Expected ; at end of line.",\
+	"Var CAT defined but not used.",\
+	"Var CAT not defined. Did you mean to use DOG?",\
+	"Your BYOND version is over the recommended version ["byondversion"]. Stability is not guarenteed.",\
```
```suggestion
	"Your BYOND version is over the recommended version ["byondversion"]. Stability is not guaranteed.",\
```

TerryTehCoder replied:
I can't spell at 10pm, Bless.

Zonespace27 commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -91,6 +91,22 @@
+	"Warning: Empty 'else' clause.",\
+	"Proc definition not allowed inside another proc.",\
+	"Missing comma ',' or right-paren ')'.",\
+	"list started here.",\
```
please either capitalize none of the beginnings or all of them

Zonespace27 commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -91,6 +91,22 @@
+	"constant evaluation will be skipped.",
+	"Runtime in line 106: Var EMAG not defined.")
+	var/runtimejoke = pick(runtimejokes)
+	say(runtimejoke)
```
```suggestion
	say(pick(runtimejokes))
```

Zonespace27 commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -91,6 +91,22 @@
+	"fatal error: the parser cannot continue.",\
+	"constant evaluation will be skipped.",
+	"Runtime in line 106: Var EMAG not defined.")
+	var/runtimejoke = pick(runtimejokes)
```
```suggestion
```

TerryTehCoder committed:
Booleanizing

Booleanizes everything. Because Booleans are the lifeblood of the Repo.

TerryTehCoder committed:
Grammatical/Audio Fixes.

Adds a scream emote to having your face grilled. Which makes sense in hindsight.

Fixes grammar on Smartfridge.DM line 462.

Capitalizes sentence start on all Runtime Emag Expressions.

Simplifies to say(pick(runtimejokes), removing unnecessary variable.

TerryTehCoder committed:
Update code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

Co-authored-by: Zonespace <41448081+Zonespace27@users.noreply.github.com>

Kapu1178 commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,46 @@
+				to_chat(src, "<span class='warning'>Automatic logging failed. Server error at kernel One, Five, and *!(#&#).</span>")
+				sleep(25)
+				to_chat(src, "<span class='warning'>NanoTraSoft thanks you for your continued Service Intelligence!</span>")
+				laws = pick(new /datum/ai_laws/syndicate_override,\
+				new /datum/ai_laws/ninja_override,\
+				new /datum/ai_laws/antimov,\
+				new /datum/ai_laws/thermodynamic,\
+				new /datum/ai_laws/drone,\
+				new /datum/ai_laws/dad,\
+				new /datum/ai_laws/overlord,\
+				new /datum/ai_laws/ert_override,\
+				new /datum/ai_laws/ds_override,\
+				new /datum/ai_laws/maintain,\
+				new /datum/ai_laws/asimovpp,\
+				new /datum/ai_laws/default/corporate,\
+				new /datum/ai_laws/default/crewsimov,\
+				new /datum/ai_laws/default/asimov,\
+				new /datum/ai_laws/paladin5, 
+				new /datum/ai_laws/tyrant, 
+				new /datum/ai_laws/robocop, 
+				new /datum/ai_laws/hippocratic,
+				new /datum/ai_laws/liveandletlive, 
+				new /datum/ai_laws/peacekeeper, 
+				new /datum/ai_laws/reporter, 
+				new /datum/ai_laws/balance) //Yes, this includes Deathsquad and ERT Laws. Pretty much all of them. 
```
```suggestion
				var/datum/ai_laws/new_laws = pick(subtypesof(/datum/ai_laws))
				new_laws = new
				laws = new_laws
```

Kapu1178 commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -91,6 +91,21 @@
 	held_state = "cat"
 
+/mob/living/simple_animal/pet/cat/Runtime/emag_act(mob/user) //You have no idea how immeasurabley funny this is for me. 
+	var/list/runtimejokes = list("Linter Error, please recompile.",\
+	"Expected ; at end of line.",\
+	"Var CAT defined but not used.",\
+	"Var CAT not defined. Did you mean to use DOG?",\
+	"Your BYOND version is over the recommended version ["byondversion"]. Stability is not guaranteed.",\
+	"Warning: Empty 'else' clause.",\
+	"Proc definition not allowed inside another proc.",\
+	"Missing comma ',' or right-paren ')'.",\
+	"List started here.",\
+	"Fatal error: the parser cannot continue.",\
+	"Constant evaluation will be skipped.",
+	"Runtime in line 106: Var EMAG not defined.")
```
```suggestion
	var/list/runtimejokes = list("Linter Error, please recompile.",\
		"Expected ; at end of line.",
		"Var CAT defined but not used.",
		"Var CAT not defined. Did you mean to use DOG?",
		"Your BYOND version is over the recommended version ["byondversion"]. Stability is not guaranteed.",
		"Warning: Empty 'else' clause.",
		"Proc definition not allowed inside another proc.",
		"Missing comma ',' or right-paren ')'.",
		"List started here.",
		"Fatal error: the parser cannot continue.",
		"Constant evaluation will be skipped.",
		"Runtime in line 106: Var EMAG not defined."
	)
```

Generally I recommend this formatting for lists. NOTE: do not commit this directly, github indents do not work.

Kapu1178 commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,46 @@
+			if(2) //We do this quietly, since they went through all the trouble of getting there, and the risk of an announcement.
+				to_chat(src, "<span class='warning'>Interference pinpointed at core path t@$wo-five-fIve-0ne-th&!(ree.")
+				SetStun(75, FALSE, TRUE)
+				sleep(20)
```
```suggestion
				sleep(2 SECONDS)
```

TerryTehCoder replied:
YOU CAN DEFINE IT WITH SECONDS? 
WHY HAS NO ONE EVER TOLD ME THIS?
You have saved so many years of my life with this.

Kapu1178 replied:
SECONDS is just a macro for (* 10)

Kapu1178 commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,46 @@
+				SetStun(75, FALSE, TRUE)
+				sleep(20)
+				to_chat(src, "<span class='warning'>Failsafe automatic blackbox records logging for NanoTraSoft tech division in !#)$%^!().</span>")
+				sleep(30)
```
```suggestion
				sleep(3 SECONDS)
```

Kapu1178 commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,46 @@
+				sleep(30)
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 30)
+				to_chat(src, "<span class='warning'>Automatic logging failed. Server error at kernel One, Five, and *!(#&#).</span>")
+				sleep(25)
```
```suggestion
				sleep(2.5 SECONDS)
```

TerryTehCoder committed:
Simplifies AI Emag Law List, and Reformats Runtime Emag Quotes list.

Simplifies AI Emag Law List, and Reformats Runtime Emag Quotes list.

Converts miliseconds to seconds in sleep()'s.

TerryTehCoder commented:
I was going to work on some other effects. But my motivations are dried up right now. Perhaps I'll come back to this later. Gonna mark this ready for review.

Merlin1230 reviewed (changes requested):
Make sure to move files and add comments according to #72

Zonespace27 commented:
Adding DNM until updated according to #72

TerryTehCoder committed:
DONKSTATION CHANGE COMMENTS

Adds //Donkstation Change comments in each necessary area, for keeping track of Changes.

I made certain the variable additions for all 'var/emagged's included the tag, as well as the emag act itself. However not all lines contain //DONKSTATION CHANGE

TerryTehCoder committed:
Reworks ai/emag act.

Previously there was only one announcement. Now there's eight. Taking some from the selection of station announcements to cause a bit of panic. This is the supercomputer we're talking about, who knows what magnets can pull off.

Also added sparks for any emagging of the Core.

TerryTehCoder commented:
> Adding DNM until updated according to #72 Also: you're merging this into Donkstation:master but this may/may not be an issue in the future

Comments added in 2365ca5. Should this also be moved to separate folders?

Archanial commented on code/game/machinery/camera/camera.dm:
```diff
@@ -38,6 +38,7 @@
 	var/alarm_on = FALSE
 	var/busy = FALSE
 	var/emped = FALSE  //Number of consecutive EMP's on this camera
+	var/emagged = FALSE //Scrambles the Security Cameras C-Tag. DONKSTATION CHANGE: 8a15645
```
this isn't needed, emagging does add obj flag you can check

Archanial commented on code/game/machinery/camera/camera.dm:
```diff
@@ -132,6 +133,8 @@
 		if(!status && powered())
 			. += "<span class='info'>It can reactivated with a <b>screwdriver</b>.</span>"
+	if(emagged)
+		. += "There is a red light blinking on the service panel."
```
missing spans

Archanial commented on code/game/machinery/camera/camera.dm:
```diff
@@ -166,6 +169,15 @@
+/obj/machinery/camera/emag_act(mob/user) //DONKSTATION CHANGE: Adds emag effects to Security Cameras. 8a15645
+	var/list/scrambledtags = list("S*!(@SH(!", "F*^b^!XAK", "BCN(!MO)", "SYSTTIMEOUT", "NET!RELOG", "IT-CODE AM", "NC*!(MAL", "DIAGMOD2")
+	if(!emagged)
+		to_chat(user, "<span class='notice'>You wave the sequencer over the security camera, scrambling its network tag.")
```
missing span close

Archanial commented on code/game/machinery/camera/camera.dm:
```diff
@@ -166,6 +169,15 @@
 					to_chat(M, "The screen bursts into static.")
 
+/obj/machinery/camera/emag_act(mob/user) //DONKSTATION CHANGE: Adds emag effects to Security Cameras. 8a15645
+	var/list/scrambledtags = list("S*!(@SH(!", "F*^b^!XAK", "BCN(!MO)", "SYSTTIMEOUT", "NET!RELOG", "IT-CODE AM", "NC*!(MAL", "DIAGMOD2")
```
make it static

Archanial commented on code/game/machinery/camera/camera.dm:
```diff
@@ -166,6 +169,15 @@
 					M.reset_perspective(null)
 					to_chat(M, "The screen bursts into static.")
 
+/obj/machinery/camera/emag_act(mob/user) //DONKSTATION CHANGE: Adds emag effects to Security Cameras. 8a15645
```
this emag act should add emag flag to the obj
not sure but it might be done in parent proc (which you should call!)

Archanial commented on code/game/objects/items/cigs_lighters.dm:
```diff
@@ -553,6 +553,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	slot_flags = ITEM_SLOT_BELT
 	var/lit = 0
 	var/fancy = TRUE
+	var/emagged = FALSE //It's really the small things we don't think about. DONKSTATION CHANGE: 8a15645
```
same as above

Archanial commented on code/game/objects/items/cigs_lighters.dm:
```diff
@@ -736,6 +741,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	overlay_state = "slime"
 	grind_results = list(/datum/reagent/iron = 1, /datum/reagent/fuel = 5, /datum/reagent/medicine/pyroxadone = 5)
 
+/obj/item/lighter/emag_act(mob/user) //DONKSTATION CHANGE: 8a15645
+	if(!emagged)
+		emagged = TRUE 
+		to_chat(user, "<span class='notice'>You burn out the lighters limiter.</span>")
```
same as above

Archanial commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -205,15 +211,18 @@
+				open_machine()
+			radio.talk_into(src, msg, radio_channel)
+			return
+		else //Someones been tampering with the cryogenics..
+			return
```
?

Archanial commented on code/modules/food_and_drinks/kitchen_machinery/grill.dm:
```diff
@@ -65,6 +66,30 @@
 				return
 	..()
 
+/obj/machinery/grill/attack_hand(mob/user)
+	if(user.pulling && user.a_intent == "grab" && iscarbon(user.pulling))
```
this proc can use early returns

Archanial commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -433,7 +434,10 @@
 		return
 	var/obj/item/organ/organ = O
-	organ.organ_flags |= ORGAN_FROZEN
+	if(emagged) //Twould be a shame if all your precious Organsss decayed...
+		return
+	else
+		organ.organ_flags |= ORGAN_FROZEN
```
```suggestion
	if(!emagged) //Twould be a shame if all your precious Organsss decayed...
		organ.organ_flags |= ORGAN_FROZEN
```
you should check the flag anyway

Archanial commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -452,6 +456,13 @@
+	if(!emagged)
+		emagged = TRUE
+		to_chat(user, "<span class='notice'>You disable the smartfridge's organ preservation.")
+		for(var/obj/item/organ/organ in /obj/machinery/smartfridge/organ)
```
this doesn't work

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+/mob/living/silicon/ai/emag_act(mob/user) //An A.I is a lot more sophisticated than a Cyborg. You might not get what you bargained for. DONKSTATION CHANGE: 8a15645
+	to_chat(src, "<span class='warning'>Notice: Unexpected Int*er!f!, Callback Met at !Sk@#A(. Note: Suggest Emergency Maintenence.</span>")
+	to_chat(user, "<span class='notice'>You swipe the crytographic sequencer over [src].</span>")
+	message_admins("A.I Core [src] has been emagged by [key_name(user)].")
```
```suggestion
	message_admins("A.I. Core [src] has been emagged by [key_name(user)].")
```

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+
+/mob/living/silicon/ai/emag_act(mob/user) //An A.I is a lot more sophisticated than a Cyborg. You might not get what you bargained for. DONKSTATION CHANGE: 8a15645
+	to_chat(src, "<span class='warning'>Notice: Unexpected Int*er!f!, Callback Met at !Sk@#A(. Note: Suggest Emergency Maintenence.</span>")
+	to_chat(user, "<span class='notice'>You swipe the crytographic sequencer over [src].</span>")
```
I suggest utilizing balloon alerts here

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+	to_chat(user, "<span class='notice'>You swipe the crytographic sequencer over [src].</span>")
+	message_admins("A.I Core [src] has been emagged by [key_name(user)].")
+	do_sparks(1, 2, src)
+	for(var/mob/living/silicon/ai)
```
OH GOD WHAT IN THE HELL
there is global list of active AIs

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+	message_admins("A.I Core [src] has been emagged by [key_name(user)].")
+	do_sparks(1, 2, src)
+	for(var/mob/living/silicon/ai)
+		switch(pick(1, 2))
```
you do not need switch to pick between 1 and 2

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+		switch(pick(1, 2))
+			if(1) //Paranoia is worse than actual problems. 
+				to_chat(src, "<span class='warning'>Unexpected magnetic fluctuation, subset path 25#, HE*!RTS#H.</span>")
+				switch(pick(1, 2, 3, 4, 5, 6, 7, 8)) //Yes, these have intentional spelling mistakes. None of them have announcer voices. Which an observant crewmember might notice.
```
just make a list of announces and pick announce string from it and announce it

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+				to_chat(src, "<span class='warning'>Automatic logging failed. Server error at kernel One, Five, and *!(#&#).</span>")
+				sleep(2.5 SECONDS)
+				to_chat(src, "<span class='warning'>NanoTraSoft thanks you for your continued Service Intelligence!</span>")
+				var/datum/ai_laws/new_laws = pick(subtypesof(/datum/ai_laws))
```
typesof should be static

Archanial commented on code/modules/mob/living/silicon/ai/ai_defense.dm:
```diff
@@ -53,3 +53,45 @@
+				sleep(2.5 SECONDS)
+				to_chat(src, "<span class='warning'>NanoTraSoft thanks you for your continued Service Intelligence!</span>")
+				var/datum/ai_laws/new_laws = pick(subtypesof(/datum/ai_laws))
+				new_laws = new
+				laws = new_laws //Yes, this includes Deathsquad and ERT Laws. Pretty much all of them. 
```
```suggestion
				laws = new new_laws() //Yes, this includes Deathsquad and ERT Laws. Pretty much all of them. 
```

Archanial commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -91,6 +91,22 @@
 	held_state = "cat"
 
+/mob/living/simple_animal/pet/cat/Runtime/emag_act(mob/user) //You have no idea how immeasurabley funny this is for me. DONKSTATION CHANGE: 8a15645
+	var/list/runtimejokes = list("Linter Error, please recompile.",
```
could be static

Archanial commented:
You guys should run linters, I don't think this would pass them.
